### PR TITLE
evtgen: fix mac build and version 2.0.0 with pythia >= 8.304.

### PR DESCRIPTION
**Pull request by iarspider**
The patch is backwards-compatible. Also fix MacOS build.

### CODE REVIEW

**Comment by iarspider** [comment]
Ping @vvolkl (maintainer)

**Comment by vvolkl** [comment]
I think you have to placate flake8, otherwise lgtm. Thanks!

File: var/spack/repos/builtin/packages/evtgen/package.py
Comment (vvolkl) [inline var/spack/repos/builtin/packages/evtgen/package.py:58]: Actually could you add a comment here why this is needed? This is not really self-explanatory

File: var/spack/repos/builtin/packages/evtgen/package.py
Comment (vvolkl) [inline var/spack/repos/builtin/packages/evtgen/package.py]: Does this still work with pythia < 8.304? If not the `when` should be updated

Comment (vvolkl) [reply var/spack/repos/builtin/packages/evtgen/package.py]: Ah, apologies, I didn't see the `#if` above, that's fine then.

**Comment by vvolkl** [comment]
Version 2.01.00 also already works with newer pythia, I'll add it (once the mysterious tarball extraction issue I posted on slack is figured out)

File: var/spack/repos/builtin/packages/evtgen/package.py
Comment (iarspider) [reply var/spack/repos/builtin/packages/evtgen/package.py:58]: @vvolkl this comes from CMS experiment, it is needed for OS X build according to comments.

**Comment by iarspider** [comment]
@spackbot restart pipeline

File: var/spack/repos/builtin/packages/evtgen/package.py
Comment (vvolkl) [reply var/spack/repos/builtin/packages/evtgen/package.py:58]: Could you add those comments in the recipe? I think it will help with maintenance.

Comment (iarspider) [reply var/spack/repos/builtin/packages/evtgen/package.py:58]: Done

**Review by vvolkl — DISMISSED** [review]

File: var/spack/repos/builtin/packages/evtgen/package.py
Comment (sethrj) [reply var/spack/repos/builtin/packages/evtgen/package.py]: Wouldn't it be easier (and safer, and avoid the extra `patch` for versions that natively work) to change this to `when='@02.00.00 ^pythia@8.304:'` and have your file simply replace `Pythia8::ParticleDataEntry*` with `Pythia8::ParticleDataEntryPtr`?

Comment (iarspider) [reply var/spack/repos/builtin/packages/evtgen/package.py]: @sethrj what do you mean by "safer"? Anyway, this is only needed for 02.00.00, the patch was accepted upstream and included in 02.01.00

Comment (sethrj) [reply var/spack/repos/builtin/packages/evtgen/package.py]: Safer by the time-honored principle "if it ain't broke don't fix it" 😉 . I still think it would be better add the `^pythia@8.304:` to avoid the unnecessary patch step and longer spec name, even though this will be increasinly irrelevant (since both versions will be out of date).

**Review by sethrj — APPROVED** [review]

**Pull request merged**
